File: core/src/galenscovell/oregontrail/OregonTrailMain.java
package galenscovell.oregontrail;

import com.badlogic.gdx.Game;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import galenscovell.oregontrail.ui.screens.*;
import galenscovell.oregontrail.util.ResourceManager;

public class OregonTrailMain extends Game {
    public SpriteBatch spriteBatch;
    public AbstractScreen loadingScreen, mainMenuScreen, gameScreen;
    
    @Override
    public void create () {
        this.spriteBatch = new SpriteBatch();
        this.loadingScreen = new LoadScreen(this);
        this.mainMenuScreen = new MainMenuScreen(this);
        setScreen(loadingScreen);
    }

    @Override
    public void dispose() {
        mainMenuScreen.dispose();
        if (gameScreen != null) {
            gameScreen.dispose();
        }
        ResourceManager.dispose();
    }

    public void newGame() {
        this.gameScreen = new GameScreen(this);
    }

    public void loadGame() {

    }
}

File: core/src/galenscovell/oregontrail/OregonTrailMain.scala
package galenscovell.oregontrail

import com.badlogic.gdx.Game
import com.badlogic.gdx.graphics.g2d.SpriteBatch
import galenscovell.oregontrail.ui.screens._
import galenscovell.oregontrail.util.ResourceManager


class OregonTrailMain extends Game {
  var spriteBatch: SpriteBatch = null
  var loadingScreen: AbstractScreen = null
  var mainMenuScreen: AbstractScreen = null
  var gameScreen: AbstractScreen = null


  def create(): Unit =  {
    this.spriteBatch = new SpriteBatch
    this.loadingScreen = new LoadScreen(this)
    this.mainMenuScreen = new MainMenuScreen(this)
    setScreen(loadingScreen)
  }

  override def dispose(): Unit =  {
    mainMenuScreen.dispose()
    if (gameScreen != null) {
      gameScreen.dispose()
    }
    ResourceManager.dispose()
  }

  def newGame(): Unit =  {
    this.gameScreen = new GameScreen(this)
  }

  def loadGame(): Unit = {

  }
}
